### PR TITLE
[FARM-310] Integrate general Alert My plot

### PR DESCRIPTION
**Pull request by davidfranquet**
[FARM-310]

**NOTE: Missing the Delete profile Alert because it's in progress**

#### 📲 What

Added the general alert on
- Begin Crop
- Complete a stage
- Revert a stage
- Complete a crop
- Delete a crop

#### 🤔 Why
		
It's needed 
		
#### 🛠 How
		
Used the general Alert widget created before

#### 👀 See
		
ezgif com-video-to-gif-4

		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/model/bloc/plot/StageToStageCardViewModel.dart [hunk 1]
```diff
@@ -6,18 +6,44 @@ import 'package:farmsmart_flutter/ui/common/stage_card.dart';
...
+
+  static completeAction() => 'Completed';
+
+  static revertAction() => 'Revert to In Progress';
+
+  static readyAction() => 'Begin Stage';
+
+  static upcomingAction() => 'Please complete previous';
+
+  static inProgressAction() => 'Mark as complete';
```
Comment (LeeMatthewHiggins) [inline lib/model/bloc/plot/StageToStageCardViewModel.dart]: missing the Intl.message.  ?

Comment (davidfranquet) [reply lib/model/bloc/plot/StageToStageCardViewModel.dart]: lol yes (Monday 😅 ) 
thanks!

Comment (davidfranquet) [reply lib/model/bloc/plot/StageToStageCardViewModel.dart]: Done

**Review by LeeMatthewHiggins — APPROVED** [review]
LGTM

**Review by albertmiro — APPROVED** [review]

**Pull request merged**